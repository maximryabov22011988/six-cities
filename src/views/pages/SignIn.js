import React from 'react';
import PropTypes from 'prop-types';

import Page from '../components/Page';
import Header from '../components/Header';
import Logo from '../components/Logo';
import Content from '../components/Content';

import SignInForm from './signIn/SignInForm';

const propTypes = {
  onSignIn: PropTypes.func.isRequired,
};

function SignIn({ onSignIn }) {
  return (
    <Page page="login">
      <Header logo={<Logo position="header" />} />
      <Content page="login">
        <SignInForm onSignIn={onSignIn} />
      </Content>
    </Page>
  );
}

SignIn.propTypes = propTypes;

export default SignIn;
