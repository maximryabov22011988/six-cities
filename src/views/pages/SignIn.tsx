import * as React from 'react';

import Page from '../components/Page';
import Header from '../components/Header';
import Logo from '../components/Logo';
import Content from '../components/Content';

import SignInForm from './signIn/SignInForm';

import { onSignIn } from '../interfaces';

interface Props {
  onSignIn: onSignIn,
}

function SignIn({ onSignIn }: Props) {
  return (
    <Page parentClassName="login">
      <Header logo={<Logo position="header" />} />
      <Content parentClassName="login">
        <SignInForm onSignIn={onSignIn} />
      </Content>
    </Page>
  );
}

export default SignIn;
