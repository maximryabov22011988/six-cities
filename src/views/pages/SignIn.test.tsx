import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import SignIn from './SignIn';

it('SignIn renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SignIn onSignIn={jest.fn()} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
