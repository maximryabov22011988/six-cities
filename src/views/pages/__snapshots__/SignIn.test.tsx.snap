// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders correctly with mandatory props 1`] = `
<div
  className="page page--gray page--login"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          />
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--login"
  >
    <div
      className="container page__login-container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          className="login__form form"
          method="post"
          onSubmit={[Function]}
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden form__label"
            >
              E-mail
            </label>
            <input
              className="login__input form__input"
              maxLength={120}
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden form__label"
            >
              Password
            </label>
            <input
              className="login__input form__input"
              minLength={3}
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <button
            className="login__submit form__submit button"
            disabled={true}
            type="submit"
          >
            Sign in
          </button>
        </form>
      </section>
    </div>
  </main>
</div>
`;
